src/OpenLoco/Vehicles/CreateVehicle.cpp

```diff
@@ -817,6 +817,14 @@ namespace OpenLoco::Vehicles
     {
         GameCommands::setExpenditureType(ExpenditureType::VehiclePurchases);
         _backupVeh0 = reinterpret_cast<VehicleHead*>(-1);
+
+        const auto* company = CompanyManager::get(CompanyManager::getControllingId());
```

Comment:
I completely missed this you want to get the updating id that is what the game command is being run against https://github.com/OpenLoco/OpenLoco/blob/7e80a7a776efec044217a4a06fe1bbd6859e4f6b/src/OpenLoco/GameCommands/GameCommands.cpp#L462